atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/async/query/GetQuerySpec.java

```diff
@@ -47,40 +48,46 @@
             + "ORDER BY column1, column2 ASC "
             + "LIMIT 1;";
 
-    @org.immutables.value.Value.Auxiliary
-    public abstract ByteBuffer row();
-
-    @org.immutables.value.Value.Auxiliary
-    public abstract ByteBuffer column();
+    @Override
+    public String formatQueryString() {
+        return String.format(
+                QUERY_FORMAT,
+                cqlQueryContext().keySpace(),
+                AbstractKeyValueService.internalTableName(cqlQueryContext().tableReference()));
+    }
 
-    @org.immutables.value.Value.Auxiliary
-    public abstract long humanReadableTimestamp();
+    @Override
+    public Statement makeExecutableStatement(PreparedStatement preparedStatement) {
+        return preparedStatement.bind()
+                .setBytes("row", getQueryParameters().row())
+                .setBytes("column", getQueryParameters().column())
+                .setLong("timestamp", getQueryParameters().queryTimestamp());
+    }
 
     @Override
     public ConsistencyLevel queryConsistency() {
         return ConsistencyLevel.LOCAL_QUORUM;
     }
 
     @Override
-    public final Supplier<RowStreamAccumulator<Optional<Value>>> rowStreamAccumulatorFactory() {
+    public Supplier<RowStreamAccumulator<Optional<Value>>> rowStreamAccumulatorFactory() {
         return GetQueryAccumulator::new;
     }
 
-    @Override
-    public String formatQueryString() {
-        return String.format(QUERY_FORMAT, keySpace(), AbstractKeyValueService.internalTableName(tableReference()));
-    }
+    @org.immutables.value.Value.Auxiliary
+    public abstract GetQueryParameters getQueryParameters();
 
-    @Override
-    public Statement makeExecutableStatement(PreparedStatement preparedStatement) {
-        return preparedStatement.bind()
-                .setBytes("row", row())
-                .setBytes("column", column())
-                .setLong("timestamp", queryTimestamp());
-    }
+    @org.immutables.value.Value.Immutable
+    public interface GetQueryParameters {
+        ByteBuffer row();
```

Comment:
How about com.palantir.conjure.java.lib.Bytes? It has a zero-cost converter to a read-only ByteBuffer

Reply:
This library isn't used anywhere else and is not a project dependency. Can I add it? @felixdesouza @jeremyk-91

Reply:
why not?

Reply:
+ if you have it like this, you force consumers to know how to serialise things into a form you should know about.

Either make this contain `Cell` and you convert when you are making the executable statement or  you add a builder method here e.g.

```java
public interface GetQueryParameters {
  ...
  public static class Builder extends ImmutableGetQueryParameters.Builder {
    public Builder cell(Cell cell) {
      return this.row(...);
          .column(...);
    }
  } 
}

Reply:
I'm using `Cell` as a parameter and I prepare everything internally.